wasm/wazeroir/signature.go

```diff
@@ -0,0 +1,376 @@
+			t = wasmValueTypeToSignless(f.Signature.InputTypes[index])
+		} else {
+			t = wasmValueTypeToSignless(f.LocalTypes[index-inputLen])
+		}
+		return &signature{in: []SignLessType{t}}, nil
+	case wasm.OptCodeLocalTee:
+		inputLen := uint32(len(f.Signature.InputTypes))
+		if l := f.NumLocals + inputLen; index >= l {
+			return nil, fmt.Errorf("invalid local index for local.get %d >= %d", index, l)
+		}
+		var t SignLessType
+		if index < inputLen {
+			t = wasmValueTypeToSignless(f.Signature.InputTypes[index])
+		} else {
+			t = wasmValueTypeToSignless(f.LocalTypes[index-inputLen])
+		}
+		return &signature{in: []SignLessType{t}, out: []SignLessType{t}}, nil
+	case wasm.OptCodeGlobalGet:
+		if len(f.ModuleInstance.Globals) <= int(index) {
+			return nil, fmt.Errorf("invalid global index for global.get %d >= %d", index, len(f.ModuleInstance.Globals))
+		}
+		return &signature{
+			out: []SignLessType{wasmValueTypeToSignless(f.ModuleInstance.Globals[index].Type.ValType)},
+		}, nil
+	case wasm.OptCodeGlobalSet:
+		if len(f.ModuleInstance.Globals) <= int(index) {
+			return nil, fmt.Errorf("invalid global index for global.get %d >= %d", index, len(f.ModuleInstance.Globals))
+		}
+		return &signature{
+			in: []SignLessType{wasmValueTypeToSignless(f.ModuleInstance.Globals[index].Type.ValType)},
+		}, nil
+	case wasm.OptCodeI32Load:
+		return signature_I32_I32, nil
+	case wasm.OptCodeI64Load:
+		return signature_I32_I64, nil
+	case wasm.OptCodeF32Load:
+		return signature_I32_F32, nil
+	case wasm.OptCodeF64Load:
+		return signature_I32_F64, nil
+	case wasm.OptCodeI32Load8s, wasm.OptCodeI32Load8u, wasm.OptCodeI32Load16s, wasm.OptCodeI32Load16u:
+		return signature_I32_I32, nil
+	case wasm.OptCodeI64Load8s, wasm.OptCodeI64Load8u, wasm.OptCodeI64Load16s, wasm.OptCodeI64Load16u,
+		wasm.OptCodeI64Load32s, wasm.OptCodeI64Load32u:
+		return signature_I32_I64, nil
+	case wasm.OptCodeI32Store:
+		return signature_I32I32_None, nil
+	case wasm.OptCodeI64Store:
+		return signature_I32I64_None, nil
+	case wasm.OptCodeF32Store:
+		return signature_I32F32_None, nil
+	case wasm.OptCodeF64Store:
+		return signature_I32F64_None, nil
+	case wasm.OptCodeI32Store8:
+		return signature_I32I32_None, nil
+	case wasm.OptCodeI32Store16:
+		return signature_I32I32_None, nil
+	case wasm.OptCodeI64Store8:
+		return signature_I32I64_None, nil
+	case wasm.OptCodeI64Store16:
+		return signature_I32I64_None, nil
+	case wasm.OptCodeI64Store32:
+		return signature_I32I64_None, nil
+	case wasm.OptCodeMemorySize:
+		return signature_None_I32, nil
+	case wasm.OptCodeMemoryGrow:
+		return signature_I32_I32, nil
+	case wasm.OptCodeI32Const:
+		return signature_None_I32, nil
+	case wasm.OptCodeI64Const:
+		return signature_None_I64, nil
+	case wasm.OptCodeF32Const:
+		return signature_None_F32, nil
+	case wasm.OptCodeF64Const:
+		return signature_None_F64, nil
+	case wasm.OptCodeI32eqz:
+		return signature_I32_I32, nil
+	case wasm.OptCodeI32eq, wasm.OptCodeI32ne, wasm.OptCodeI32lts,
```

Comment:
I'm guessing that due to number of ops we aren't creating constants of each op including their signature.. if were not burdensome I might just because code navigation could easily link to the signature which could be helpful elsewhere when we are thinking about the signature implications of an op. For example, here we can do a reference search and then look in the case statement to figure it out. That's good also, though more indirect vs having the signature a field of the op